### DISCUSSION TITLE
Moretti Design - Status

### DISCUSSION THREAD

**Discussion by foxmt** [philibertc/micronova_controller#61] [Q&A]
I just received the card with the electronic equipment. I soldered the components and proceeded with the configuration.

I have tested on my Morreti Design stove but I am not getting any status or temperature information.

I have tested with the "test" files, but some files seem to have some problems when compiling. I was forced to correct.
Only the test1 ON/OFF seems to work.

An idea of the problem ?

Here is my log

```
*wm:Connecting to SAVED AP: WIFI
*wm:AutoConnect: SUCCESS 
*wm:STA IP Address: 192.168.1.1
user
pass
Attempting MQTT connection...connected
Creating topics for HomeAssistant
HomeAssistant topics created.
```

**Comment by foxmt** [comment]
I put some debug messages in the code. I'm stuck on the step

while (StoveSerial.available())

**Comment by foxmt** [comment]
Whats the difference between Serial.available method and OnOff ? I don't understand why it works on with on/off test but not reading informations.

**Reply by philibertc** [reply]
Hello,
while(Serial.available) waits to receive the two bytes of information. The loop does not stop until the information is received. So if it hangs here, it means that nothing is received.

With on/off test we do not expect any value so nothing is stuck.

**Comment by foxmt** [comment]
I'm debugging with test2_read

When I'm setting `digitalWrite(ENABLE_RX, LOW);` before reading, I'm getting some params in serial reading

Param=cc value=52 0
Param=fd value=3 0
Param=c4 value=60 0
Param=df value=33 0
Param=ff value=1 0
Param=c2 value=62 0

**Reply by foxmt** [reply]
I'm getting this only when the stove is on. When it's off, there's nothing

**Reply by philibertc** [reply]
Where do you put `digitalWrite(ENABLE_RX, LOW)`?

What is happening is strange, the on or off state of the stove should not influence the reading.

**Reply by foxmt** [reply]
In test2 file

Just before reading serial (StoveSerial.available()))
```
    int sensorReading = 0; // variable to store the value read from the sensor pin
    uint8_t rxCount = 0;
    stoveRxData[0] = 0x00;
    stoveRxData[1] = 0x00;
    **digitalWrite(ENABLE_RX, LOW);**
    while (StoveSerial.available()) //It has to be exactly 2 bytes, otherwise it's an error
    {
        stoveRxData[rxCount] = StoveSerial.read();
        rxCount++;
    }
    digitalWrite(ENABLE_RX, HIGH);
    if (rxCount == 2)
```

**Reply by philibertc** [reply]
Okay, this shouldn't change anything because there is already `digitalWrite(ENABLE_RX, LOW);` before each call of the `checkStoveReply();` function.
Maybe this is a faulty contact?

**Reply by foxmt** [reply]
Ok i will test with the 2nd card

**Reply by philibertc** [reply]
Ah in fact it can not be a false contact in the wires, because the RX and TX is on the same cable.
But you can still test with the other card, we never know.

**Reply by foxmt** [reply]
I tested the 2nd card, there is no difference.

However, I had indicated the wrong information. It is not the addition of LOW before the reading that allows the reading, but the deactivation of the HIGH just before rxCount

**Comment by foxmt** [comment]
Solved by making a hardware modification as explained in this topic

https://github.com/philibertc/micronova_controller/issues/31#issuecomment-2710487294
